index.js

```diff
@@ -0,0 +1,98 @@
+const express = require('express');
+const knex = require('knex')({
+    client: 'mysql',
+    connection: {
+        host: '127.0.0.1',
+        port: 3306,
+        user: 'root',
+        password: 'my-secret-pw',
+        database: 'test'
+    }
+});
+
+const mealsRouter = require('./routers/meals');
+const reservationsRouter = require('./routers/reservations');
+
+const app = express();
+const port = 3000;
+
+const sendJSON = (res, status, data) => {
+    res.status(status).json(data);
+};
+
+app.use(express.json());
+
+app.use('/api/meals', mealsRouter);
+app.use('/api/reservations', reservationsRouter);
+
+app.get('/test-connection', async (req, res) => {
+    try {
+        const result = await knex.raw("SELECT 1 + 1 AS result");
+        sendJSON(res, 200, { message: "Database connection is working!", result: result[0] });
+    } catch (error) {
+        console.error("Error testing database connection:", error);
+        sendJSON(res, 500, { error: "Database connection failed." });
+    }
+});
+
+app.get('/future-meals', async (req, res) => {
+    try {
+        const futureMeals = await knex.raw("SELECT * FROM meal WHERE when > NOW()");
+        sendJSON(res, 200, futureMeals[0].length ? futureMeals[0] : []);
+    } catch (error) {
+        console.error("Error retrieving future meals:", error);
+        sendJSON(res, 500, { error: "An error occurred while retrieving future meals." });
+    }
+});
+
+app.get('/past-meals', async (req, res) => {
+    try {
+        const pastMeals = await knex.raw("SELECT * FROM meal WHERE when < NOW()");
+        sendJSON(res, 200, pastMeals[0].length ? pastMeals[0] : []);
+    } catch (error) {
+        console.error("Error retrieving past meals:", error);
+        sendJSON(res, 500, { error: "An error occurred while retrieving past meals." });
+    }
+});
+
+app.get('/all-meals', async (req, res) => {
+    try {
+        const allMeals = await knex.raw("SELECT * FROM meal ORDER BY id ASC");
+        sendJSON(res, 200, allMeals[0].length ? allMeals[0] : []);
+    } catch (error) {
+        console.error("Error retrieving all meals:", error);
+        sendJSON(res, 500, { error: "An error occurred while retrieving all meals." });
+    }
+});
+
+app.get('/first-meal', async (req, res) => {
+    try {
+        const firstMeal = await knex.raw("SELECT * FROM meal ORDER BY id ASC LIMIT 1");
+        if (firstMeal[0].length === 0) {
+            sendJSON(res, 404, { message: "No meals found." });
+        } else {
+            sendJSON(res, 200, firstMeal[0][0]);
```

Comment:
same comment as above:
`can allMeals be null or undefined, then you would need to handle this case also :)`